### PR TITLE
[Cleanup] silence warnings

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Fixes noisy warning I get since upgrading to Kubuntu 18.10 and accordingly to gcc 8.2.
The first one is a bit dubious - an alternative would be to increase the filename buffers, but I understand that this is supposed to generate MS-DOS 8.3 filenames!?

### Changes made:

Silence warnings.

### CODE REVIEW

File: core/uemis-downloader.c [hunk 1]
```diff
@@ -583,7 +593,7 @@ static bool uemis_get_answer(const char *path, char *request, int n_param_in,
 		if (import_thread_cancelled)
 			return false;
 		progress_bar_fraction = filenr / (double)UEMIS_MAX_FILES;
-		snprintf(fl, 13, "ANS%d.TXT", filenr - 1);
+		snprintf(fl, 13, "ANS%d.TXT", sanitize_filenr(filenr - 1));
```
Comment (neolit123) [inline core/uemis-downloader.c]: how about turning this in a utility function:
```
void write_filenr(char *buf, int bufsz, int filenr) 
{
    filenr = filenr < 0 ? 0 : filenr % 10000;
    snprintf(buf, bufsz, "ANS%d.TXT", filenr);    
}

// usage
write_filenr(fl, sizeof(fl), filenr - 1)
```

Comment (bstoeger) [reply core/uemis-downloader.c]: Yes, fine by me. The question remains: do we actually want to code around compiler warnings?

Comment (dirkhh) [reply core/uemis-downloader.c]: I'd like the `write_filenr()` helper better...

**Review by dirkhh — APPROVED** [review]
I like this in general - but I think having a helper function to actually just write that ANS file would be more natural.

**Comment by bstoeger** [comment]
Alright - had a go at putting this in an extra function. Problem: I have no device to test!

**Review by dirkhh — APPROVED** [review]

**Pull request merged**